src/NHibernate.Test/TestDialect.cs

```diff
@@ -55,6 +58,11 @@ public TestDialect(Dialect.Dialect dialect)
 		/// </summary>
 		public virtual bool SupportsEmptyInserts => true;
 
+		/// <summary>
+		/// Supports condition not bound to any data, like "where @p1 = @p2".
+		/// </summary>
+		public virtual bool SupportsNonDataBoundCondition => true;
```

Comment:
Copied from #703, I will have to resolve a conflict on one of those PR depending on merge order. But I was needing this.